tests/deprecation/tests.py

```diff
@@ -170,3 +172,14 @@ class Deprecated(DeprecatedMixin, RenamedMixin, Renamed):
                 '`DeprecatedMixin.old` is deprecated, use `new` instead.',
                 '`RenamedMixin.old` is deprecated, use `new` instead.',
             ])
+
+
+class DeprecationInstanceCheckTest(SimpleTestCase):
+    def test_warning(self):
+        class Manager(six.with_metaclass(DeprecationInstanceCheck)):
+            alternative = 'fake.path.Foo'
+            deprecation_warning = RemovedInNextVersionWarning
+
+        msg = '`Manager` is deprecated, use `fake.path.Foo` instead.'
+        with self.assertRaisesMessage(RemovedInNextVersionWarning, msg):
+            isinstance(object, Manager)
```

Comment:
Didn't know `__instancecheck__` was never called when `instance.__class__ is self`.